nav2_collision_monitor/src/velocity_polygon.cpp

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2022 Dexory
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/velocity_polygon.hpp"
+#include "nav2_collision_monitor/polygon.hpp"
+
+#include "nav2_util/node_utils.hpp"
+
+
+namespace nav2_collision_monitor
+{
+
+VelocityPolygon::VelocityPolygon(
+  const nav2_util::LifecycleNode::WeakPtr & node,
+  const std::string & polygon_name,
+  const std::string & velocity_polygon_name,
+  const std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  const std::string & base_frame_id,
+  const tf2::Duration & transform_tolerance)
+: node_(node), polygon_name_(polygon_name), velocity_polygon_name_(velocity_polygon_name),
+  tf_buffer_(tf_buffer), base_frame_id_(base_frame_id), transform_tolerance_(transform_tolerance)
+{
+  RCLCPP_INFO(logger_, "[%s]: Creating VelocityPolygon", velocity_polygon_name_.c_str());
+}
+
+
+VelocityPolygon::~VelocityPolygon()
+{
+  RCLCPP_INFO(logger_, "[%s]: Destroying VelocityPolygon", velocity_polygon_name_.c_str());
+  poly_.clear();
+}
+
+
+bool VelocityPolygon::getParameters()
+{
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  try {
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".points",
+      rclcpp::PARAMETER_DOUBLE_ARRAY);
+    std::vector<double> polygon_points = node->get_parameter(
+      polygon_name_ + "." + velocity_polygon_name_ + ".points").as_double_array();
+
+    if (!Polygon::setPolygonShape(polygon_points, poly_)) {
+      RCLCPP_ERROR(
+        logger_,
+        "[%s][%s]: Polygon has incorrect points description",
+        polygon_name_.c_str(), velocity_polygon_name_.c_str());
+      return false;
+    }
```

Comment:
Why aren't you setting this as a static footprint mode?

Reply:
How are you deciding if its a static footprint mode vs if a message has come in yet on the topic for the subscription's footprint to be useful?

Reply:
Added the similar try catch pattern as in the `polygon` class for waiting the subscription topic. In current design, the sub polygon will always expose the topic for reconfiguring, do we want to skip the topic subscription if `points` are provided in the param file?